apps/src/feedback.js

```diff
@@ -502,6 +529,23 @@ FeedbackUtils.prototype.displayFeedback = function (options, requiredBlocks,
   }
 };
 
+/**
+ * Converts the image data uri to a blob, then saves it to the server as the
+ * thumbnail for the current project.
+ * @param {string} image Image encoded as a data URI.
+ * @returns {Promise} A promise which resolves if successful.
+ */
+FeedbackUtils.saveThumbnail = function (image) {
+  return dataURIToBlob(image)
+    .then(project.saveThumbnail)
+    .then(() => {
+      return new Promise(resolve => {
+        // Save the thumbnail url to the server.
+        project.save(resolve);
```

Comment:
You can drop a layer of nesting here.
```js
.then(() => new Promise(resolve => {
  promise.save(resolve);
}));
```
Or it would be _really_ nice to wrap `project.save` in a promise-friendly wrapper - or even make it return a promise itself?
```js
const saveProject = () => new Promise(resolve => project.save(resolve));
//...
  .then(project.saveThumbnail)
  .then(saveProject);
```